pycytominer/cyto_utils/sqlite/convert.py

```diff
@@ -0,0 +1,456 @@
+        Chunk size for unique join key datasets.
+        Note: adjust to resource capabilities of
+        machine and provided dataset. Smaller
+        chunksizes may mean greater time duration
+        and lower memory consumption
+    pq_filename: str:  (Default value = "combined")
+        Target parquet filename to be used as an in-process
+        prefix and also as the resulting file from
+        the flow.
+
+    Returns
+    -------
+    str
+        Single parquet filename of file which contains
+        all SQLite data based on the outcome of a
+        Prefect flow.
+
+    """
+
+    # build a prefect flow with explicit storage based on parameter
+    with Flow("flow_convert_sqlite_to_parquet", storage=flow_storage) as flow:
+
+        # set flow parameters
+        param_sql_engine = Parameter("sql_engine", default="")
+        param_sql_tbl_basis = Parameter("sql_tbl_basis", default=sql_tbl_basis)
+        param_sql_join_keys = Parameter("sql_join_keys", default=sql_join_keys)
+        param_sql_chunk_size = Parameter("sql_chunk_size", default=sql_chunk_size)
+        param_pq_filename = Parameter("pq_filename", default=pq_filename)
+
+        # form prefect tasks from sqlite meta utils
+        task_engine_from_str = task(engine_from_str)
+        task_collect_columns = task(collect_columns)
+
+        # build an engine which may be used by other tasks
+        engine = task_engine_from_str(sql_engine=param_sql_engine)
+
+        # gather sql column and table data flow operations
+        column_data = task_collect_columns(sql_engine=engine)
+
+        # chunk the dicts so as to create batches
+        chunk_dicts = sql_select_distinct_join_chunks(
+            sql_engine=engine,
+            table_name=param_sql_tbl_basis,
+            join_keys=param_sql_join_keys,
+            chunk_size=param_sql_chunk_size,
+        )
+
+        # map to gather our concatted/merged pd dataframes as a list within this flow
+        pq_files = table_concat_to_parquet.map(
+            sql_engine=unmapped(engine),
+            column_data=unmapped(column_data),
+            prepend_tablename_to_cols=unmapped(True),
+            avoid_prepend_for=unmapped(param_sql_join_keys),
+            chunk_list_dicts=chunk_dicts,
+            filename=unmapped(param_pq_filename),
+        )
+
+        # reduce to single pq "concatted" file from the map result list
+        reduced_pq_result = multi_to_single_parquet(
+            pq_files=pq_files, filename=param_pq_filename
+        )
+
+    # run the flow
+    flow_state = flow.run(
+        executor=flow_executor,
+        parameters=dict(
+            sql_engine=str(sql_engine.url),
+            sql_tbl_basis=sql_tbl_basis,
+            sql_join_keys=sql_join_keys,
+            sql_chunk_size=sql_chunk_size,
+            pq_filename=pq_filename,
+        ),
+    )
+
+    # return the result of reduced_pq_result, a path to the new file
+    return flow_state.result[reduced_pq_result].result
```

Comment:
This looks complicated. How complex is the `flow_state` object? Will we ever need to access other elements of it?

Reply:
Thanks for calling this forward. The flow_state object is somewhat complex - it contains state and result data for all tasks within the Prefect flow. The return here was an effort to make the primary goal of the Prefect flow of practical use for a Pycytominer developer (instead of sharing all tasks / meta along the way). Because it's a new convention for the repo, I wasn't certain about much to lean towards flexibility vs convenience.

Potential changes: We could return `flow_state` on it's own here to rely on the developer to navigate the results (and allow them greater flexibility with what happens to the flow). Another further step in this direction might be to deliver a flow without running it, enriching options in this territory as well. Thoughts?

Additional information which may help: Prefect 1.x, used for changes related to this PR, is stable and works well. There are also ongoing efforts for Prefect 2.x which change how flows are handled (flows are formed through decorators for Python functions, similar to how `@task` is used for these changes). This may mean that we eventually end up closer to the "potential changes" structure anyways, when Prefect 2.x is out of beta, and if this is something you'd like to head towards.